File: app/src/main/java/com/my/televip/features/ghostMode/HideTyping.java
package com.my.televip.features.ghostMode;

import com.my.televip.Class.ClassLoad;
import com.my.televip.Class.ClassNames;
import com.my.televip.Clients.ClientManager;
import com.my.televip.obfuscate.ArgsResolver;
import com.my.televip.obfuscate.Obfuscate;

public class HideTyping {

    public static boolean isTypingRequest(Object object) {
        if (!ClientManager.isTgnetObfuscated()) {
            String className = object.getClass().getName();
            return className.contains("TL_messages_setTyping") ||
                    className.contains("TL_messages_setEncryptedTyping");
        } else {
            Class<?> objectClass = object.getClass();
            return objectClass.equals(ClassLoad.getClass(Obfuscate.getClassName(ClassNames.TL_MESSAGES_SET_TYPING))) ||
                    objectClass.equals(ClassLoad.getClass(Obfuscate.getClassName(ClassNames.TL_MESSAGES_SET_ENCRYPTED_TYPING)));
        }
    }
}

File: app/src/main/java/com/my/televip/features/ghostMode/HideTyping.kt
package com.my.televip.features.ghostMode

import com.my.televip.clazz.ClassLoad
import com.my.televip.clazz.ClassNames
import com.my.televip.clients.ClientManager
import com.my.televip.obfuscate.Obfuscate

object HideTyping {

    @JvmStatic
    fun isTypingRequest(objectParam: Any): Boolean {
        return if (!ClientManager.isTgnetObfuscated()) {
            val className = objectParam.javaClass.name
            className.contains("TL_messages_setTyping") ||
                    className.contains("TL_messages_setEncryptedTyping")
        } else {
            val objectClass = objectParam.javaClass
            objectClass == ClassLoad.getClass(Obfuscate.getClassName(ClassNames.TL_MESSAGES_SET_TYPING)) ||
                    objectClass == ClassLoad.getClass(Obfuscate.getClassName(ClassNames.TL_MESSAGES_SET_ENCRYPTED_TYPING))
        }
    }
}
